org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/DebugAdapter.java

```diff
@@ -101,10 +102,6 @@ public DebugAdapter(BiConsumer<Events.DebugEvent, Boolean> consumer, IProviderCo
                     responseBody = attach(JsonUtils.fromJson(arguments, Requests.AttachArguments.class));
                     break;
 
-                case "restart":
-                    responseBody = restart(JsonUtils.fromJson(arguments, Requests.RestartArguments.class));
-                    break;
-
```

Comment:
Not really. it just means we don't support restart request in debug session, but support the old restart policy (disconnect first and relaunch).